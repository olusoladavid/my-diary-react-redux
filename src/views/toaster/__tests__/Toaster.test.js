import thunk from "redux-thunk";
import configureStore from "redux-mock-store";

import ConnectedToaster, { Toaster } from "../Toaster";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const storeState = {
  toaster: {
    visible: false,
    toasts: []
  }
};

const props1 = {
  visible: true,
  toasts: [{ type: "error", message: "error" }],
  hideToast: jest.fn()
};

const props2 = {
  visible: false,
  toasts: [{ type: "progress", message: "progress" }],
  hideToast: jest.fn()
};

it("renders without crashing when visible", () => {
  const wrapper = shallow(<Toaster {...props1} />);
  expect(wrapper).toMatchSnapshot();
});

it("does not render when not visible", () => {
  const wrapper = shallow(<Toaster {...props2} />);
  expect(wrapper).toMatchSnapshot();
});

it("should synchronise with app state in the store", () => {
  const store = mockStore(storeState);
  const wrapper = shallow(<ConnectedToaster {...props1} store={store} />);
  expect(wrapper.prop("visible")).toBe(false);
});
