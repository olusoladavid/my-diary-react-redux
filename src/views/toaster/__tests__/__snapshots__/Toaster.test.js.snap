// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render when not visible 1`] = `""`;

exports[`renders without crashing when visible 1`] = `
<div
  className="toast-container js-toast-container"
>
  <div
    className="toast-container__close-btn js-toast-close"
    onClick={[MockFunction]}
  >
    Ã—
  </div>
  <div
    className="toast js-toast toast--error"
    key="0"
  >
    <span
      className="toast__text js-toast-text"
    >
      error
    </span>
  </div>
</div>
`;
